lib/Sema/CSSimplify.cpp

```diff
@@ -1535,31 +1535,29 @@ ConstraintSystem::matchTupleTypes(TupleType *tuple1, TupleType *tuple2,
   return getTypeMatchSuccess();
 }
 
-// Determine whether conversion is allowed between two function types
-// based on their representations.
+/// Check where a representation is a subtype of another.
+///
+/// The subtype relationship is defined as:
+/// 1. any representation R is a sub-type of itself.
+/// 2. a thin representation is a subtype of any other representation.
+/// 3. a thick representation is a subtype of any other representation.
```

Comment:
Is this true? Because thick-to-thin subtype is not allowed ... sorry, maybe I'm just a bit confused :)

Reply:
I think you are correct, at least that is how it was handled before, even examples still have passing a swift closure to @conversion(c) parameter as an error.

Reply:
Yeah, made a mistake there, it should be "a thick representation is a subtype of any other thick representation"

Reply:
Sounds good!

Reply:
Ah makes sense :)

Reply:
Question, do you think would be informational note sub-type relation and variance here? As in `(@convetion(c) () -> Void) -> Void  sub (@convention(block) () -> Void) -> Void`? I'm just wondering if this could be helpful in someway

Reply:
Function subtyping has multiple aspects: (1) representation/convention (2) variance (3) throws (4) `@escaping`. At the moment, this PR only documents (1), because that's the trickiest part.

The example you're describing falls out as a composition of (1) + (2). Are you suggesting that I document both (1) and (2) separately? Or is the suggestion that I specifically document that "(1) + (2) can be combined in such a way"?

Reply:
>  this PR only documents (1), because that's the trickiest part

Ah sure, better thinking about I agree it makes sense to only focus on (1) here
I just mentioned because (1) + (2) combined used to get me confused =]